packages/server/src/fhircast/routes.ts

```diff
@@ -71,65 +71,43 @@ publicSTU3Routes.get('/.well-known/fhircast-configuration', (_req: Request, res:
 });
 
 // Register a new subscription
+// Or publish an event depending on payload
+// See: https://hl7.org/fhir/uv/fhircast/2024May/2-6-RequestContextChange.html#:~:text=Similar%20to%20the%20Hub%E2%80%99s%20notifications%20to%20the%20Subscriber%2C%20the%20Subscriber%20MAY%20request%20context%20changes%20with%20an%20HTTP%20POST%20to%20the%20hub.url
 protectedCommonRoutes.post(
   '/',
-  [
-    body('hub.channel.type').notEmpty().withMessage('Missing hub.channel.type'),
-    body('hub.mode').notEmpty().withMessage('Missing hub.mode'),
-    body('hub.topic').notEmpty().withMessage('Missing hub.topic'),
-    body('hub.events').notEmpty().withMessage('Missing hub.events'),
-  ],
+  oneOf(
+    [
+      [
+        body('id').notEmpty().withMessage('Missing event ID'),
+        body('timestamp').notEmpty().withMessage('Missing event timestamp'),
+        body('event').notEmpty().withMessage('Missing event payload'),
+        body('event["hub.topic"]').notEmpty().withMessage('Missing event["hub.topic"]'),
+        body('event["hub.event"]').notEmpty().withMessage('Missing event["hub.event"]'),
+        body('event.context').notEmpty().withMessage('Missing event.context'),
+      ],
+      [
+        body('hub.channel.type').notEmpty().withMessage('Missing hub.channel.type'),
+        body('hub.mode').notEmpty().withMessage('Missing hub.mode'),
+        body('hub.topic').notEmpty().withMessage('Missing hub.topic'),
+        body('hub.events').notEmpty().withMessage('Missing hub.events'),
+      ],
+    ],
+    { errorType: 'least_errored' }
```

Comment:
`least_errored` error type means of the different groups, the one with the least errors is selected as our "choice", ie. the one for which the validation feedback is given back to the user in the case that neither group passes validation